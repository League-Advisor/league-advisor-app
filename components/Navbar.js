import React from 'react'
import Link from 'next/link'
import Image from 'next/image'


export default function Navbar() {
    return (
        <>
            <nav className="sticky top-0 z-0 flex-col justify-between w-2/12 rounded-md h-10/12">
                <div className="h-full bg-black">
                    <div className="flex justify-center py-20 pr-4 shadow-sm">
                        <Image src="/logo.png" width="140" height="140" alt="logo" />

                    </div>
                    <div className="pl-10">
                        <ul className="pt-8 space-y-12">
                            {/* Profile */}
                            <li className="flex items-center space-x-4 text-2xl font-bold text-white cursor-pointer hover:text-blue-600">
                                <svg xmlns="http://www.w3.org/2000/svg" className="w-12 h-12" fill="none" viewBox="0 0 24 24"
                                    stroke="currentColor">
                                    <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2"
                                        d="M8 5H6a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2v-1M8 5a2 2 0 002 2h2a2 2 0 002-2M8 5a2 2 0 012-2h2a2 2 0 012 2m0 0h2a2 2 0 012 2v3m2 4H10m0 0l3-3m-3 3l3 3" />
                                </svg>
                                <Link href="/profile" >Profile</Link>
                            </li>
                            {/* Dashboard */}
                            <li className="flex items-center space-x-4 text-2xl font-bold text-white cursor-pointer hover:text-blue-600">
                                <svg xmlns="http://www.w3.org/2000/svg" className="w-12 h-12 " fill="none" viewBox="0 0 24 24"
                                    stroke="currentColor">
                                    <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2"
                                        d="M9 17V7m0 10a2 2 0 01-2 2H5a2 2 0 01-2-2V7a2 2 0 012-2h2a2 2 0 012 2m0 10a2 2 0 002 2h2a2 2 0 002-2M9 7a2 2 0 012-2h2a2 2 0 012 2m0 10V7m0 10a2 2 0 002 2h2a2 2 0 002-2V7a2 2 0 00-2-2h-2a2 2 0 00-2 2" />
                                </svg>

                                <Link href="/">Dashboard</Link>

                            </li>
                            {/* Patch Notes */}
                            <li className="flex items-center space-x-4 text-2xl font-bold text-white cursor-pointer hover:text-blue-600">
                                <svg xmlns="http://www.w3.org/2000/svg" className="w-12 h-12" fill="none" viewBox="0 0 24 24"
                                    stroke="currentColor">
                                    <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2"
                                        d="M13 10V3L4 14h7v7l9-11h-7z" />
                                </svg>
                                <Link href="/patchNotes">Patch Notes</Link>
                            </li>
                            {/* About Us */}
                            <li className="flex items-center space-x-4 text-2xl font-bold text-white cursor-pointer hover:text-blue-600">
                                <svg xmlns="http://www.w3.org/2000/svg" className="w-12 h-12" fill="none" viewBox="0 0 24 24"
                                    stroke="currentColor">
                                    <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2"
                                        d="M5 8h14M5 8a2 2 0 110-4h14a2 2 0 110 4M5 8v10a2 2 0 002 2h10a2 2 0 002-2V8m-9 4h4" />
                                </svg>
                                <Link href="/aboutUs">About Us</Link>
                            </li>
                            {/* Logout */}
                            <li className="flex items-center space-x-4 text-2xl font-bold text-white cursor-pointer hover:text-blue-600">
                                <svg xmlns="http://www.w3.org/2000/svg" className="w-12 h-12" fill="none" viewBox="0 0 24 24" stroke="currentColor">
                                    <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2"
                                        d="M17 16l4-4m0 0l-4-4m4 4H7m6 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h4a3 3 0 013 3v1" />
                                </svg>
                                <Link href="/login">Logout</Link>
                            </li>
                        </ul>
                    </div>


                </div>


            </nav>
        </>
    )
}