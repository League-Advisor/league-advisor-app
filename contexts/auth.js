import { createContext, useContext, useState } from 'react';
import jwt from 'jsonwebtoken';
import axios from 'axios'
const baseUrl = process.env.NEXT_PUBLIC_API_URL;
const tokenUrl = baseUrl + '/api/token/';
import { useEffect } from "react";

const AuthContext = createContext();

export function useAuth() {
    const auth = useContext(AuthContext);
    if (!auth) throw new Error('You forgot AuthProvider!');
    return auth;
}

export function AuthProvider(props) {

    const [state, setState] = useState({
        tokens: null,
        user: null,
        login: login,
    });

    async function login(username, password) {

        const response = await axios.post(tokenUrl, { username, password });

        const decodedAccess = jwt.decode(response.data.access);

        const newState = {
            tokens: response.data,
            user: {
                id: decodedAccess.user_id,
                username: decodedAccess.username,
                email: decodedAccess.email,
                summoner_name: decodedAccess.summoner_name,
                profile_icon: decodedAccess.profile_icon,
                summoner_server: decodedAccess.summoner_server,
                summoner_level: decodedAccess.summoner_level,
                summoner_rank: decodedAccess.summoner_rank,
                summoner_champion_mastery: decodedAccess.summoner_champion_mastery,
                summoner_match_history: decodedAccess.summoner_match_history,
            },

        }

        localStorage.setItem("token", JSON.stringify(newState))
        setState(prevState => ({ ...prevState, ...newState }));
    }

    useEffect(() => {
        const data = localStorage.getItem("token")
        if (data) {
            setState(JSON.parse(data))
        }
    }, {})

    return (
        <AuthContext.Provider value={state}>
            {props.children}
        </AuthContext.Provider>
    );
}