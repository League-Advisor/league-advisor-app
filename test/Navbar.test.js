import { render, getByText, getByTestId } from '@testing-library/react'
import '@testing-library/jest-dom'
import * as React from 'react'
import Navbar from '../components/Navbar'
import { AuthProvider } from '../contexts/auth'



const Providers = ({ children }) => {
    return children;
};

const customRender = (ui, options = {}) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };


describe("Navbar", () => {
    it("render profile", () => {
        render(<AuthProvider ><Navbar /></AuthProvider>);
        const { asFragment } = render(<button>Profile</button>);
        const fragmentElement = asFragment().firstChild;
        const a = getByText(fragmentElement, 'Profile');
        expect(a).not.toBe(null);


    });
});

describe("Navbar", () => {
    it("svg test", () => {
        render(<AuthProvider ><Navbar /></AuthProvider>);
        expect(getByTestId(document.documentElement, "icon"),).toBeInTheDocument();

    });
});

describe("Navbar", () => {
    it("unorder list", () => {
        render(<AuthProvider ><Navbar /></AuthProvider>);
        expect(getByTestId(document.documentElement, "list"),).toBeInTheDocument();

    });
});
