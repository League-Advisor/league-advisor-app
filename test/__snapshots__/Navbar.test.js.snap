// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar name 1`] = `
<a>
  Profile
</a>
`;

exports[`Navbar render nav 1`] = `
<a>
  Profile
</a>
`;

exports[`Navbar render navigation bar 1`] = `
<a>
  Profile
</a>
`;

exports[`Navbar render profile 1`] = `
<a>
  Profile
</a>
`;
